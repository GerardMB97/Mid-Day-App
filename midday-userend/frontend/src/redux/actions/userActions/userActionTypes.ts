const userActionTypes = {
  SIGN_UP: 'SIGN_UP',
  SIGN_IN: 'SIGN_IN',
  RESET_STATUS: 'RESET_STATUS'
};

export default userActionTypes;
