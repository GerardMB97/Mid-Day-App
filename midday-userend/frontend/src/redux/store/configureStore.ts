import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import rootReducer from '../reducers';
import initialState from './initialState';

function configureStore () {
  const composeEnhancers = compose;

  return createStore(
    rootReducer,
    initialState,
    composeEnhancers(applyMiddleware(reduxImmutableStateInvariant(), thunk))
  );
}

export default configureStore();
