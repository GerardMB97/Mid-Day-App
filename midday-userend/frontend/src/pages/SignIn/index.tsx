import colors from '../../../colors';
import React, { useEffect } from 'react';
import { ImageBackground, View, StyleSheet, Image, TextInput, TouchableOpacity, Text } from 'react-native';
import { Navigation } from '../../models';
import { bindActionCreators, Dispatch } from 'redux';
import { signIn } from '../../redux/actions/userActions/userActions';
import Modal from '../../components/Modal';
import modal from '../../constants/modalText';
import { connect } from 'react-redux';

const styles = StyleSheet.create({
  image: {
    flex: 1
  },
  container: {
    marginTop: 100,
    width: '100%',
    flex: 1,
    alignItems: 'center'
  },
  logo: {
    width: 400,
    height: 250
  },
  input: {
    width: 250,
    height: 40,
    borderRadius: 5,
    paddingLeft: 10,
    backgroundColor: 'white',
    marginTop: 40

  },
  button: {
    width: 100,
    height: 40,
    borderRadius: 5,
    backgroundColor: colors.green,
    marginTop: 40,
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 20
  },
  text: {
    color: 'white'
  },
  modal: {
    position: 'absolute',
    top: 350,
    zIndex: 2
  }
});
function SignIn ({ navigation, user, actions }:{navigation:Navigation, user: any, actions: any}) {
  const [email, setEmail] = React.useState('');
  const [pwd, setPwd] = React.useState('');
  const [wrongUser, setWrongUser] = React.useState(false);

  const handleModal = (setter: Function) => {
    setter(true);
    setTimeout(() => { setter(false); }, 4000);
  };
  useEffect(() => {
    if (user.status === 1) {
      handleModal(setWrongUser);
    } else if (user.status === 2) {
      navigation.navigate('LandingPage');
    }
    console.log(user.status);
  }, [user]);
  return (
    <ImageBackground style={styles.image} source={{ uri: 'https://trello-attachments.s3.amazonaws.com/6041f773bf2ba60154c38447/417x626/b5a3b5bc578c8c8a9603c72b18f9670e/comida-ingredientes_1220-4884.jpg' }}>
      <View style={styles.container}>
        <Image style={styles.logo} source = {{ uri: 'https://trello-attachments.s3.amazonaws.com/6041f773bf2ba60154c38447/886x624/909520ea82dfc7447a5fe02c2583c808/logo_blanco2_%282%29.png' }}></Image>
        <TextInput value={email} onChangeText={text => setEmail(text)} style={styles.input} placeholder="Introduzca su correo"></TextInput>
        {wrongUser && <View style={styles.modal}><Modal modalText={modal.wrongUser} ></Modal></View>}
        <TextInput secureTextEntry={true} value={pwd} onChangeText={text => setPwd(text)} style={styles.input} placeholder="Introduzca su contraseña"></TextInput>
        <TouchableOpacity onPress={() => { actions.signIn(email, pwd); }} style={styles.button} ><Text>Sign In</Text></TouchableOpacity>
        <TouchableOpacity onPress={() => { navigation.navigate('SignUp'); }}><Text style={styles.text}>Todaviá no estás registrado? Pincha aquí.</Text></TouchableOpacity>
      </View>
    </ImageBackground>
  );
}

function mapStateToProps ({ user }:{user:any}) {
  return { user };
}

function mapDispatchToProps (dispatch: Dispatch) {
  return { actions: bindActionCreators({ signIn }, dispatch) };
}

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);
