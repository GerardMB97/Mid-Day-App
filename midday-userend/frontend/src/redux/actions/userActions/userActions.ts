import axios from 'axios';
import userActionTypes from './userActionTypes';

import { SignUpRoute, SignInRoute, updateUserIsNew, deleteBookingRoute, deleteInvitationRoute, updateBookingPaxRoute, updateAllergiesRoute, newBookingRoute, bookingToRestRoute, bookingToUserRoute } from '../../../constants/dataBase';
import { Dispatch } from 'redux';
import { addInvitation } from '../../../utils';
export const signUp = (name:string, email:string, password:string) => {
  return async (dispatch:Dispatch) => {
    try {
      const { data } = await axios.post(SignUpRoute, { name, email, password });
      dispatch({
        type: userActionTypes.SIGN_UP,
        data
      });
    } catch (error) {
    }
  };
};

export const signIn = (email: string, password: string) => {
  return async (dispatch:Dispatch) => {
    try {
      const { data } = await axios.post(SignInRoute, { email, password });

      dispatch({
        type: userActionTypes.SIGN_IN,
        data
      });
    } catch (error) {
      dispatch({
        type: userActionTypes.SIGN_IN,
        data: null
      });
    }
  };
};

export const updateIsnew = ({ _id }:{_id:string}) => {
  return async (dispatch:Dispatch) => {
    try {
      const { data } = await axios.put(updateUserIsNew, { _id });
      dispatch({
        type: userActionTypes.UPDATE_ISNEW,
        data
      });
    } catch (error) {

    }
  };
};

export const updateUserAllergies = (allergen: string) => {
  return {
    type: userActionTypes.UPDATE_USER_ALLERGIES,
    allergen
  };
};

export const updateAllergiesDB = ({ _id, allergies }:any) => {
  return async (dispatch:Dispatch) => {
    try {
      const { data } = await axios.put(updateAllergiesRoute, { _id, allergies });

      dispatch({
        type: userActionTypes.UPDATE_ALLERGIES_DB,
        user: data
      });
    } catch (error) {

    }
  };
};

export const deleteBooking = ({ email }: {email:string}, bookingId:string) => {
  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${deleteBookingRoute}${email}`, { bookingId });

    dispatch({
      type: userActionTypes.DELETE_BOOKING,
      user: data
    });
  };
};

export const deleteInvitation = (userId:string, invitationId:string) => {
  return async (dispatch:Dispatch) => {
    try {
      await axios.put(updateBookingPaxRoute, { userId, bookingId: invitationId });
      const { data } = await axios.put(deleteInvitationRoute, { userId, invitationId });
      dispatch({
        type: userActionTypes.DELETE_INVITATION,
        user: data
      });
    } catch (error) {

    }
  };
};

export const createBooking = (date, hour, bookingAdmin, pax, people, restaurantId) => {
  return async (dispatch:Dispatch) => {
    try {
      const booking = {
        date,
        hour,
        bookingAdmin,
        pax,
        people,
        restaurant: restaurantId

      };
      const { data } = await axios.post(newBookingRoute, booking);

      data.people.forEach(async (person) => { if (bookingAdmin !== person.user) { await addInvitation(person.user, data._id); } });
      const reqBody = {
        bookingId: data._id,
        restaurantId

      };
      await axios.put(bookingToRestRoute, reqBody);
      const userReqBody = {
        bookingId: data._id,
        userId: bookingAdmin
      };
      const user = await axios.put(bookingToUserRoute, userReqBody);
      dispatch({
        type: userActionTypes.CREATE_BOOKING,
        user: user.data
      });
    } catch (error) {

    }
  };
};
