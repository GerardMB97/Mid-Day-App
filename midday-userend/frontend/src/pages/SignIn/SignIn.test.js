import React from 'react';
import { fireEvent, render } from 'react-native-testing-library';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import * as actions from '../../redux/actions/userActions/userActions';
import SignIn from '.';

jest.mock('../../redux/actions/restaurantActions/restaurantAction');

describe('Given a SignIn component', () => {
  const mockStore = configureStore();
  const store = mockStore({ user: {} });
  jest.spyOn(actions, 'signIn').mockReturnValue({ type: '' });

  describe('When rendered', () => {
    test('Then it should render properly', () => {
      const tree = render(<Provider store={store}><SignIn></SignIn></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });
});
