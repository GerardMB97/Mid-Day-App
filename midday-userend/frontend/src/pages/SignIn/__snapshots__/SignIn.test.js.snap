// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a SignIn component When rendered Then it should render properly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://trello-attachments.s3.amazonaws.com/6041f773bf2ba60154c38447/417x626/b5a3b5bc578c8c8a9603c72b18f9670e/comida-ingredientes_1220-4884.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 100,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://trello-attachments.s3.amazonaws.com/6041f773bf2ba60154c38447/886x624/909520ea82dfc7447a5fe02c2583c808/logo_blanco2_%282%29.png",
        }
      }
      style={
        Object {
          "height": 250,
          "width": 400,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Introduzca su correo"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 40,
          "marginTop": 40,
          "paddingLeft": 10,
          "width": 250,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Introduzca su contraseña"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 40,
          "marginTop": 40,
          "paddingLeft": 10,
          "width": 250,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(135, 175, 147)",
          "borderRadius": 5,
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginTop": 40,
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <Text>
        Sign In
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Todaviá no estás registrado? Pincha aquí.
      </Text>
    </View>
  </View>
</View>
`;
